src/main/java/org/neo4j/junit/jupiter/causal_cluster/Neo4jCluster.java

```diff
@@ -35,15 +38,23 @@
 public interface Neo4jCluster {
 
 	/**
+	 * This method is guaranteed to return the URI of a core server and never a replica.
+	 *
 	 * @return An URI into this cluster.
 	 */
-	URI getURI();
+	default URI getURI() {
+
+		Collection<URI> allCoreUris = getURIs();
+		int n = ThreadLocalRandom.current().nextInt(allCoreUris.size());
+		return allCoreUris.stream().skip(n).findFirst().get();
+	}
```

Comment:
should we add a method here that takes a type?

```
default URI getURI(Type serverType) {
```

Reply:
I would leave this to the user. And probably prefer a `getServers(Predicate<Neo4jServer> p)` for that.